packages/trader/src/App/Components/Elements/VideoStream/volume-control.tsx

```diff
@@ -0,0 +1,177 @@
+
+            return;
+        }
+        setIsMuted(false);
+        toggleMute(false);
+    };
+
+    const buttonClickHandler = () => {
+        if (is_muted) {
+            volume_bar_ref?.current && volume_bar_ref.current.style.setProperty('height', `${(volume ?? 0.5) * 100}%`);
+            toggleMute(false);
+        } else {
+            volume_bar_ref?.current && volume_bar_ref.current.style.setProperty('height', '0%');
+            toggleMute(true);
+        }
+        setIsMuted(prev => !prev);
+    };
+
+    const mouseMoveHandler = (e: React.MouseEvent<HTMLElement, MouseEvent>) => {
+        e.preventDefault();
+        e.stopPropagation();
+
+        if (!is_dragging.current) return;
+        if (!volume_bar_ref?.current) return;
+
+        const new_height = calculateNewHight(e);
+
+        volume_bar_ref.current.style.setProperty('transition', 'none');
+        volume_bar_ref.current.style.setProperty('height', `${new_height}%`);
+
+        onVolumeChange(new_height / 100);
+        checkVolumeControl(new_height);
+    };
+
+    const mouseUpHandler = (e: React.MouseEvent<HTMLElement, MouseEvent>) => {
+        e.preventDefault();
+        e.stopPropagation();
+
+        if (!is_dragging.current) return;
+
+        if (volume_bar_ref?.current) {
+            volume_bar_ref.current.style.setProperty('transition', 'all 0.3s linear');
+        }
+        is_dragging.current = false;
+
+        // Hide volume bar if user finished changing volume outside of the container
+        if (!(e.target as HTMLElement).closest('.player__volume-bar__wrapper')) setShowVolume(false);
+    };
+
+    const mouseDownHandler = (e: React.MouseEvent<HTMLElement, MouseEvent>) => {
+        e.preventDefault();
+        e.stopPropagation();
+
+        const point = document.querySelector('.player__volume-dot');
+        setShiftY((point?.getBoundingClientRect().bottom ?? 0) - e.clientY);
+
+        is_dragging.current = true;
+    };
+
+    const onRewind = (e: React.MouseEvent<HTMLElement, MouseEvent>) => {
+        e.preventDefault();
+        e.stopPropagation();
+
+        if ((e.target as HTMLElement).className === 'player__volume-dot') return;
+        if (!volume_bar_ref?.current) return;
+
+        const new_height = calculateNewHight(e);
+
+        volume_bar_ref.current.style.setProperty('transition', 'none');
+        volume_bar_ref.current.style.setProperty('height', `${new_height}%`);
+
+        onVolumeChange(new_height / 100);
+        checkVolumeControl(new_height);
+    };
+
+    React.useEffect(() => {
+        document.addEventListener('mousemove', mouseMoveHandler as unknown as (this: Document, ev: MouseEvent) => void);
+        document.addEventListener('mouseup', mouseUpHandler as unknown as (this: Document, ev: MouseEvent) => void);
+
+        return () => {
+            document.removeEventListener(
+                'mousemove',
+                mouseMoveHandler as unknown as (this: Document, ev: MouseEvent) => void
```

Comment:
we can remove all assertions here, the correct type of event should be declared for the callbacks (mouseMoveHandler, mouseUpHandler, etc.)